webapp/components/Modal/ReportModal.vue

```diff
@@ -54,6 +91,12 @@ export default {
       const name = this.$filters.truncate(this.name, 30)
       return this.$t(`report.${this.type}.message`, { name })
     },
+    formSchema() {
+      const validReportSchema = validReport({ translate: this.$t })
+      return {
+        ...validReportSchema.formSchema,
+      }
+    },
   },
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-10-14T19:51:11Z" title="Monday, October 14th 2019, 9:51:11 pm +02:00">Oct 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -40,10 +62,55 @@ export default {
     id: { type: String, required: true },
   },
   data() {
+    let valuesReasonCategoryOptions = [
+      'discrimination_etc',
+      'pornographic_content_links',
+      'glorific_trivia_of_cruel_inhuman_acts',
+      'doxing',
+      'intentional_intimidation_stalking_persecution',
+      'advert_products_services_commercial',
+      'criminal_behavior_violation_german_law',
+      'other',
+    ]
+
+    let reasonCategoryOptions = []
+    valuesReasonCategoryOptions.forEach(reasonCategory => {
+      reasonCategoryOptions.push({
+        label: this.$t('report.reason.category.options.' + reasonCategory),
+        value: reasonCategory,
+      })
+    })
+
     return {
       isOpen: true,
       success: false,
       loading: false,
+      failsValidations: true,
+      form: {
+        reasonCategory: null,
+        reasonCategoryOptions,
+        reasonDescription: '',
+      },
+      formSchema: {
+        reasonCategory: {
+          type: 'enum',
+          required: true,
+          validator: (rule, value, callback, source, options) => {
+            this.form.reasonCategory = value
```


Why would you update the form model through a validator :thinking: ?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-15T18:03:05Z" title="Tuesday, October 15th 2019, 8:03:05 pm +02:00">Oct 15, 2019</time>_

---


don't know, has been updated